src/backend/access/transam/slru.c

```diff
@@ -762,7 +762,12 @@ SimpleLruDownloadSegment(SlruCtl ctl, int pageno, char const* path)
 	}
 	segno = pageno / SLRU_PAGES_PER_SEGMENT;
 
-	buffer = palloc(BLCKSZ * SLRU_PAGES_PER_SEGMENT);
+	if (!strcmp(GetConfigOption("neon.use_communicator_worker", true, false), "on")) {
+		buffer = NULL;
+	} else {
+		buffer = palloc(BLCKSZ * SLRU_PAGES_PER_SEGMENT);
+	}
+
```

Comment:
Please check the boolean value of the GUC's registered variable instead of the GUC's config string.

There are various strings that will cause the GUC infrastructure to enable a boolean, as the input is matched against (case-insensitive) `true`, `on`, `yes`, `1` for truthy values.

Reply:
> Please check the boolean value of the GUC's registered variable instead of the GUC's config string.
> 
> There are various strings that will cause the GUC infrastructure to enable a boolean, as the input is matched against (case-insensitive) `true`, `on`, `yes`, `1` for truthy values.

I thought so too, but the configuration code registers a boolean option explicitly, and this `GetConfigOption` function returns only either `"on"` or `"off"` strings explicitly in case of a boolean option, so I think we are good here with the values, aren't we?

Reply:
I think it'd be fine to create the GUC's variable in Postgres, and control its values in the extension by registering the Guc with `neon.use_communicator_worker`. We've done that before, and probably will do so in the future, too, if and when needed.